lib/manageiq/loggers/journald.rb

```diff
@@ -28,12 +44,59 @@ def add(severity, message = nil, progname = nil)
         end
 
         message = formatter.call(format_severity(severity), progname, message)
+        caller_object = caller_locations.last
 
-        Systemd::Journal.print(log_level_map[severity], message)
+        # TODO: Allow the syslog facility to be set via the configuration settings.
+        Systemd::Journal.message(
+          :message           => message,
+          :message_id        => message_id,
+          :priority          => log_level_map[severity],
+          :application       => 'manageiq',
+          :syslog_identifier => determine_identifier(message),
+          :syslog_facility   => 'local3',
+          :code_line         => caller_object.lineno,
+          :code_file         => caller_object.absolute_path
+        )
       end
 
       private
 
+      # Determine the identifier based on the message. Ideally this would
+      # be set by the provider once we have pluggable loggers, but for now
+      # we parse the message.
+      #
+      def determine_identifier(message)
+        case message
+        when /Amazon/
+          'manageiq-aws'
```

Comment:
Really want to avoid this at all costs.  Would rather creating an instance of the journald logger for the different providers and setting this at initialization time

Reply:
Oh, I think I see what you mean. Something like this?

```
Systemd::Journal.new(:provider => 'azure') # Not sure if this is possible
# or
log = Systemd::Journal.new
log.provider = 'azure'
```

Reply:
Oh, it looks like we could do `Systemd::Journal.new(nil, :progname => 'manageiq-azure')`, and then we could use `progname` as the identifier.

I don't know, though, I think this would make querying the journal less friendly.

Reply:
No its not a Systemd::Journal attribute, I mean add a new `ManageIQ::Loggers::Journald` attribute and use that when logging:
```
$azure_log = ManageIQ::Loggers::Journald.new
$azure_log.syslog_identifier = "azure"
```

Then when you're logging just use that:
```
Systemd::Journal.message(
  :message           => message,
  :syslog_identifier => syslog_identifier
)
```

Would that work?

Reply:
I've updated it so that it will default to using the `progname`(the core Logger API has this feature, I wasn't aware until recently), but it's also available as an accessor. So, now you could do it either way.

```
log = ManageIQ::Loggers::Journald.new(nil, :progname => "manageiq-azure")
log.syslog_identifier # => "manageiq-azure"
# or
log = ManageIQ::Loggers::Journald.new
log.syslog_identifier = "manageiq-azure"
```
Setting the progname results in slightly nicer output, since our formatter will then include it in the message.

I also added an accessor to `syslog_facility` (defaults to `local3`), and `application` (defaults to 'manageiq`.